src/basic/env-util.c

```diff
@@ -779,7 +779,7 @@ int serialize_environment(FILE *f, char **environment) {
                 if (!ce)
                         return -ENOMEM;
 
-                fprintf(f, "env=%s\n", *e);
+                fprintf(f, "env=%s\n", ce);
```

Comment:
I'm not sure I understand why `systemd-run` unescapes `\0xa` while `systemctl` sends `\0xa` as is. In other words, I still don't understand what the difference between
```
systemd-run --property 'Environment=A=A\x0aB' ...
```
and
```
systemctl set-environment 'A=A\0xaB'
```
is. It seems that everything unescapes `\x0a` except `set-environment`.